src/include/api_inline.h

```diff
@@ -0,0 +1,53 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *  All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#pragma once
+
+/*
+ * __wt_api_track_cursor_start --
+ *     Start tracking a cursor API entry point for statistics.
+ */
+static WT_INLINE void
+__wt_api_track_cursor_start(WT_SESSION_IMPL *session)
+{
+    /*
+     * Track cursor API calls, so we can know how many are in the library at a point in time. These
+     * need to be balanced. If the api call counter is zero, it means these have been used in the
+     * wrong order compared to the other enter/end macros.
+     */
+    WT_ASSERT(session, session->id == 0 || session->api_call_counter != 0);
```

Comment:
The check against session->id == 0 is sprinkled around this change. It turns out that the default session is used for a bunch more things than it should be. I'd like to fix that and remove the checks, but not within the purview of this change - so I've just excluded it.